File: app/src/main/java/com/melean/taskstimetracker/data/database/RealmObjects/TaskRealmObject.java
package com.melean.taskstimetracker.data.database.RealmObjects;



import com.melean.taskstimetracker.data.models.TaskModel;

import io.realm.RealmObject;
import io.realm.annotations.Index;
import io.realm.annotations.Required;

public class TaskRealmObject extends RealmObject{

    @Required
    @Index
    private String taskName;


    public TaskRealmObject() {
    }

    public TaskRealmObject(String taskName) {
        this.taskName = taskName;
    }

    public String getTaskName() {
        return taskName;
    }

    public void setTaskName(String taskName) {
        this.taskName = taskName;
    }

    public static RealmObject from(TaskModel taskModel) {
        TaskRealmObject object = new  TaskRealmObject();
        object.taskName = taskModel.getTaskName();
        return object;
    }
}

File: app/src/main/java/com/melean/taskstimetracker/data/database/RealmObjects/TaskRealmObject.kt
package com.melean.taskstimetracker.data.database.RealmObjects


import com.melean.taskstimetracker.data.models.TaskModel

import io.realm.RealmObject

open class TaskRealmObject : RealmObject() {
    var taskName: String? = null

    companion object {
        fun from(taskModel: TaskModel): RealmObject {
            val model =  TaskRealmObject()
            model.taskName = taskModel.taskName
            return model
        }
    }
}
